packages/build/src/steps/plugin.js

```diff
@@ -40,7 +41,14 @@ export const firePluginStep = async function ({
     } = await callChild({
       childProcess,
       eventName: 'run',
-      payload: { event, error, envChanges, netlifyConfig, constants },
+      payload: {
+        event,
+        error,
+        envChanges,
+        featureFlags,
```

Comment:
we're providing the flags to every single plugin with this, no matter who wrote it and wether we trust it. I don't think these flags are super secret, so this should be fine - if not, is there some guard we could put in place? e.g. use the `pluginPackageJson` that's passed into the function, and check it against an allow-list?

Reply:
How much work do you think it would be to build that allow-list functionality? Just thinking from a perspective of being conservative in what we expose to plugins before we've fully settled on how we expose feature flags in general.

Reply:
but if we're only trying to hide this while figuring out the details, then that's probably a good solution.

Reply:
These flags are passed down from buildbot (not filtered or anything). This means that we will also get secret values down the road here!

Instead, we might only pass them to the "core plugins"? Which is the next runtime etc.? instead of to all build plugins?

Furthermore, we should now change in buildbot the way how we gather feature flags and use the newly added `tag` filter to only pass feature flags for `@netlify/build`.

Currently all Netlify flags are passed down (backend, frontend, etc...)

![CleanShot 2023-05-02 at 15 17 06](https://user-images.githubusercontent.com/11156362/235677793-fb6c080c-a157-4f6d-839e-55978afd4353.png)

Reply:
> Instead, we might only pass them to the "core plugins"?

Right, that's what I mean by "allow list". How would you recommend we check wether we can trust a plugin?

Reply:
I would recommend adding a property to the `pluginsList` https://list-v2--netlify-plugins.netlify.app/plugins.json

Reply:
or if the npm package starts with `@netlify/` as then it's one of our trusted npm packages